server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -3815,6 +3819,66 @@ public void testTypelessGet() throws IOException {
         closeShards(shard);
     }
 
+    public void testRecoverFromStoreReserveRetentionLeases() throws Exception {
```

Comment:
Can you move this test to IndexShardRetentionLeaseTests?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/37995/commits/f6b270c1d31cc59c211510d28d55eb2f330534f6.